src/Assertions.jl

```diff
@@ -89,6 +86,64 @@ function _global_indent()
   return s
 end
 
+@doc Markdown.doc"""
+    @vprint(S::Symbol, k::Int, msg::String)
+    @vprint S k msg
+
+    @vprint(S::Symbol, msg::String)
+    @vprint S msg
+
+This macro can be used to control printings inside the code.
+
+The macro `vprint` has two or three entry arguments: a symbol `S` refered as a
+*verbose scope*, an optional integer `k` and a string `msg`. If `k` is not
+specified, it is set by default to $1$.
+
+To each verbose scope `S` is associated a *verbose level* `l` which is cached.
```

Comment:
```suggestion
To each verbose scope `S` is associated a *verbose level* `l`.
```
What do you mean by "is cached"? Maybe that it is persistent?